share/poly.go

```diff
@@ -387,6 +387,11 @@ func (p *PubPoly) Equal(q *PubPoly) bool {
 		return false
```

Comment:
The comment above seems wrong now, it also returns false early in case of trivial mismatches on the same group.